CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift

```diff
@@ -178,3 +205,157 @@ extension PeerToPeerRequestHandler: MessageEndpointDelegate {
     }
 }
 
+private class LocalWinCustomConflictResolver: ConflictResolverProtocol {
+    func resolve(conflict: Conflict) -> Document? {
+        let localDoc = conflict.localDocument!
+        let remoteDoc = conflict.remoteDocument!
+        let docId = conflict.documentID
+        let remoteDocId = remoteDoc.id
+        let localDocId = localDoc.id
+        if remoteDocId != localDocId {
+            print("Remote docId and local docId are different")
+        }
+        if remoteDocId != docId {
+            print("Remote docId doesn't match with conflict docId")
+        }
+        if localDocId != docId {
+            print("Local docId doesn't match with conflict docId")
+        }
+        return localDoc
+    }
+}
+
+private class RemoteWinCustomConflictResolver: ConflictResolverProtocol{
+    func resolve(conflict: Conflict) -> Document? {
+        let localDoc = conflict.localDocument!
+        let remoteDoc = conflict.remoteDocument!
+        let docId = conflict.documentID
+        let remoteDocId = remoteDoc.id
+        let localDocId = localDoc.id
+        if remoteDocId != localDocId {
+            print("Remote docId and local docId are different")
+        }
+        if remoteDocId != docId {
+            print("Remote docId doesn't match with conflict docId")
+        }
+        if localDocId != docId {
+            print("Local docId doesn't match with conflict docId")
+        }
+        return remoteDoc
+    }
+}
+
+private class NullWinCustomConflictResolver: ConflictResolverProtocol{
+    func resolve(conflict: Conflict) -> Document? {
+        let localDoc = conflict.localDocument!
+        let remoteDoc = conflict.remoteDocument!
+        let docId = conflict.documentID
+        let remoteDocId = remoteDoc.id
+        let localDocId = localDoc.id
+        if remoteDocId != localDocId {
+            print("Remote docId and local docId are different")
+        }
+        if remoteDocId != docId {
+            print("Remote docId doesn't match with conflict docId")
+        }
+        if localDocId != docId {
+            print("Local docId doesn't match with conflict docId")
+        }
+        return nil
+    }
+}
+
+private class MergeWinCustomConflictResolver: ConflictResolverProtocol{
+    func resolve(conflict: Conflict) -> Document? {
+        let localDoc = conflict.localDocument!
+        let remoteDoc = conflict.remoteDocument!
+        let docId = conflict.documentID
```

Comment:
need to make sure either the local doc or the remote doc is not null

Reply:
same as above

Reply:
Added null check